Source/Task/LocklessQueue.h

```diff
@@ -0,0 +1,740 @@
+// Copyright (c) Microsoft Corporation
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#pragma once
+
+/*****************************************************************************
+ 
+ LocklessQueue is a lock free queue that can take any type as a payload.
+ Internally it maintains a demand-allocated block of contiguous nodes that
+ are used as the nodes of a linked list.  Data pushed into the queue uses a
+ copy, and popping data copies that same data out to your variable.
+ 
+ An initial block size is created up front and the size of the block can be customized
+ via the constructor.  If the queue runs out of space, it can create new blocks.  This process
+ is also lock / wait free, although it has an increassed "push" cost as the new block
+ is allocated and initialized. If there is not enough memory to create a new block
+ push_back returns false.
+ 
+ N.B. LocklessQueue MUST be either dynamically allocated or stack allocated as a
+ local variable.  It cannot be used as a member variable in a class or structure
```

Comment:
That wasn't my experience with the compiler Windows is using.  If I declared as a member of a class that had different alignment it wouldn't follow through and the alignment was off.  Maybe compiler bug?

Reply:
Thinking about it, how would the compiler preserve alignment? It would have to know how offset every instance is and magically change the padding...

Ignore me